release_docs/RELEASE.txt

```diff
@@ -191,14 +191,6 @@ Bug Fixes since HDF5-1.16.0 release
 
     Testing
     -------
-    - Disabled running of MPI Atomicity tests for OpenMPI major versions < 5
```

Comment:
This RELEASE.txt note appears in the 1.14 history and can be removed from both develop and the 1.16 branch